src/common/config.cc

```diff
@@ -874,7 +874,7 @@ int md_config_t::set_val(const char *key, const char *val, bool meta)
   }
 
   // couldn't find a configuration option with key 'key'
-  return -ENOENT;
+  return -EINVAL;
```

Comment:
it would be better if we can differentiate the EINVAL and ENOENT. if we really want to have a better error message, i'd recommend update `CephContext::do_command()` to handle `ENOENT` as a special case.